eth/block_specifier.go

```diff
@@ -0,0 +1,137 @@
+package eth
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+type BlockSpecifier struct {
+	Number           *Quantity
+	Tag              *Tag
+	Hash             *Hash
+	RequireCanonical *bool
+	Raw              bool
```

Comment:
minor: do we want to tag this as `json:"-"` just in case someone manages to get around using our custom Marshal/Unmarshal routines?

Reply:
Avoided them to keep consistency with `eth.BlockNumberOrTag` and avoid confusion wrt the recommended approach for (un)marshalling.